### PR TITLE
Update kdropdown menu api

### PR DESCRIPTION
**Pull request by marcellamaki**
## Description
This PR changes the KDropdownMenu API by addressing the issue that KDropdownMenu also contained a button. Now, KDropdownMenu is _only_ the popover elements

#### Issue addressed
Addresses #164
Does a better job fixing #136 (which caused a regression with my last fix). 
Fix the regression where the [additional options for the button (previously in a `<slot/>` in KDropdownMenu is no longer working](https://kolibri-design-system.netlify.app/buttons/#dropdowns), as you can see in the documentation

### Before/after screenshots
There are no visual changes to the UI (I hope!)

## Steps to test
I will open a corresponding kolibri PR that points to this commit next week with instructions for testing.

### At a high level, how did you implement this?
The makes `KDropdownMenu` a slot for `KButton` (or `KIconButton`) rather than responsible for also creating the button.

### Does this introduce any tech-debt items?
 Doing this would require some (fairly straightforward) remediation following a KDS version bump in Kolibri (11 instances of component are used) and KDP (5 instances of component). None in Studio.

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical and brittle code paths are covered by unit tests
- [ ] The change has been added to the `changelog`

## Reviewer guidance

- [ ] Is the code clean and well-commented?
- [ ] Are there tests for this change?
- [ ] Are all UI components LTR and RTL compliant (if applicable)?
- [ ] _Add other things to check for here_

## Post-merge updates and tracking

- [ ] Learning Platform update PR is submitted
- [ ] Learning Platform update PR is merged
- [ ] Studio update PR is submitted
- [ ] Studio update PR is merged
- [ ] Data Portal update PR is submitted
- [ ] Data Portal update PR is merged

### CODE REVIEW

**Comment by MisRob** [comment]
@marcellamaki Thank you. Marcella, could you please make sure to fix the linter check? There are some prettier issues with spacing that make previewing changes difficult so it'd be helpful for review.

**Comment by marcellamaki** [comment]
Yes - I will do this today. Thanks for pointing this out @MisRob, I'd missed it, and it definitely makes the diff seem much larger than it is!

**Comment by marcellamaki** [comment]
[The corresponding PR in Kolibri](https://github.com/learningequality/kolibri/pull/9616)

**Review by MisRob — CHANGES_REQUESTED** [review]
Thank you, @marcellamaki, we reviewed the code together with @akolson, and overall it's looking good. There were a couple of places where we asked some questions, and it's possible that no updates will be required there, only clarification.

Also, one more general question regarding the PR description, what did you mean by _"Fix the regression where the additional options for the button (previously in a <slot/> in KDropdownMenu is no longer working, as you can see in the documentation"_?


Before merging though, we'll need to make sure that the changelog and documentation are updated to reflect API updates for all `KButton`, `KIconButton`, and `KDropdownMenu`. This may not be a full list, just a pointer to a couple of places I noticed:

**1. Nothing shows up in the _Dropdowns_ section of the _Buttons and links_ page and the section text is obsolete** (http://localhost:4000/buttons#dropdowns)

Screenshot from 2022-08-23 13-59-24

**2. The _Overview_ section on the `KDropdownMenu` page is obsolete**
Screenshot from 2022-08-23 14-00-51 (http://localhost:4000/kdropdownmenu)

**3. `menu` slot needs to be documented on `KButton` and `KIconButton` pages** (http://localhost:4000/kiconbutton)

Screenshot from 2022-08-23 14-07-25

**4. I'd suggest mentioning how `KButton`, `KIconButton`, and `KDropdownMenu` can be used together including a code example, ideally. This could be, for example, done on the `KDropdownMenu` page and linked to from `KButton` and `KIconButton` pages. Or in any other way. The goal of this feedback is to make sure that everyone can understand easily how to use these components together.**

File: lib/buttons-and-links/KIconButton.vue
Comment (MisRob) [inline lib/buttons-and-links/KIconButton.vue:95]: Rather than "passes to existing 'UiTooltip' prop for position" comment, could we list values that get accepted? This will go to the documentation automatically. From the KDS user point of view, it doesn't matter that much what we use under the hood when compared to knowing what values will work with this prop. A nice example is `appearance` prop of `KExternalLink` https://design-system.learningequality.org/kexternallink/#prop:appearance

**Comment by MisRob** [comment]
This also fixes https://github.com/learningequality/kolibri-design-system/issues/67

File: lib/keen/UiMenuOption.vue
Comment (marcellamaki) [reply lib/keen/UiMenuOption.vue:18]: That is a great question, and I don't know the answer... I do think it is _technically_ doable, but I don't know if we have done it before, or if there are potential considerations that I am not thinking of. Perhaps this is something that could be discussed in an upcoming KDS tactical? I don't know if in this case it should be blocking though

File: lib/buttons-and-links/KButton.vue
Comment (marcellamaki) [reply lib/buttons-and-links/KButton.vue:29]: Previously, the component was structured to either have slots or text, but not both. My thinking was that there could be scenarios where some combination of text and slots would be necessary. For example, in Kolibri, we have one place where there is a secondary button "Options" with a dropdown menu.
Screen Shot 2022-09-01 at 3 23 25 PM
I made this change thinking about this use case, which would allow both the `text` on line 30 to remain as well as the `<slots>`. There might be a better way to do this, though!

File: lib/keen/UiMenuOption.vue
Comment (marcellamaki) [reply lib/keen/UiMenuOption.vue:29]: .... I am not sure I meant to 🙃 I will double check and follow up

**Comment by marcellamaki** [comment]
> "Fix the regression where the additional options for the button (previously in a in KDropdownMenu is no longer working, as you can see in the documentation"

@MisRob and @akolson - what I am referencing (not very clearly) is this [part of the documentation on "Buttons"](https://kolibri-design-system.netlify.app/buttons/#dropdowns) where you can see:
Screen Shot 2022-09-01 at 10 23 55 AM 
This button "ICON DROPDOWN" should actually be looking like an options menu icon ( ··· ), but due to the regression I introduced last year, it broke entirely and I failed to notice (it seems only in the documentation? I don't know of any places in kolibri where there were regressions, or at least don't recall any reports of regressions... but it is possible we missed it).

File: lib/keen/UiMenuOption.vue
Comment (MisRob) [reply lib/keen/UiMenuOption.vue:18]: I don't think it's blocking this PR, but it will be good to chat about it during the tactical

File: lib/buttons-and-links/KButton.vue
Comment (MisRob) [reply lib/buttons-and-links/KButton.vue:29]: Thank you for the explanation, I think that's fine as long as it's communicated in the documentation, which means that this comment needs to be updated as the default slot doesn't take precedence over the `text` prop anymore
```
<!-- @slot Pass sub-components into the button, which provides more flexibility than and takes precedence over the `text` prop -->
```

and it would also ideally be mentioned in the changelog.

**Comment by MisRob** [comment]
@marcellamaki I see, okay, thank you

File: lib/keen/UiMenuOption.vue
Comment (marcellamaki) [reply lib/keen/UiMenuOption.vue]: inadvertently committed (I recall I was testing). Un-commented and added back

**Comment by MisRob** [comment]
As @marcellamaki is on vacation, instead of posting feedback on the newest changes, I rebased on top of the latest `develop` and pushed two small fixes of changelog and documentation before merging.

**Review by MisRob — APPROVED** [review]

**Pull request merged**
